### PR TITLE
make commit --all mid-merge continue the merge

### PR DESCRIPTION
**Pull request by novalis**

### CODE REVIEW

File: node/lib/util/commit.js [hunk 1]
```diff
@@ -497,7 +498,8 @@ exports.commit = co.wrap(function *(metaRepo,
                                     metaStatus,
                                     message,
                                     subMessages,
-                                    noVerify) {
+                                    noVerify,
+                                    mergeParent) {
```
Comment (shijinglu) [inline node/lib/util/commit.js:502]: could you also update the doc with the new parameter, and and a short description on what happens if `mergeParent` is present?

File: node/lib/util/commit.js [hunk 2]
```diff
@@ -550,6 +558,13 @@ exports.commit = co.wrap(function *(metaRepo,
             if (headCommit !== null) {
                 parents.push(headCommit);
             }
+
+            if (mergeTree !== null) {
+                const mergeSubParent = yield mergeTree.entryByPath(name);
+                if (mergeSubParent.id() !== headCommit.id()) {
+                    parents.push(mergeSubParent.id());
```
Comment (shijinglu) [inline node/lib/util/commit.js:565]: in line 559, we just pushed a commit to `parents`, now we are pushing an object id to `parents`. Is it designed to have different types in the same array?

File: node/lib/util/commit.js [hunk 3]
```diff
@@ -597,14 +612,22 @@ exports.commit = co.wrap(function *(metaRepo,
 
     const tree = yield index.writeTree();
     const headCommit = yield metaRepo.getHeadCommit();
+    const parents = [headCommit];
+    if (mergeParent) {
+        parents.push(mergeParent);
```
Comment (shijinglu) [inline node/lib/util/commit.js:617]: what happens if `mergeParent.id() === headCommit.id()`? Will it be two parents of the same commit?

File: node/test/util/commit.js [hunk 4]
```diff
@@ -3385,3 +3385,74 @@ x=U:Cmsg\n#x-2 s=Sa:s;Bmaster=x;Os Cmsg\n#s-1 README.md=foo,addedbyhook=bar`;
...
+            yield MergeUtil.merge(repo,
+                                  null,
+                                  m4,
+                                  MergeCommon.MODE.NORMAL,
+                                  Open.SUB_OPEN_OPTION.FORCE_OPEN,
+                                  [],
+                                  "msg",
+                                  function() {});
+        } catch (e) {
+            // we expect the merge to fail, but that's OK
```
Comment (shijinglu) [inline node/test/util/commit.js:3419]: why is merge fail expected? Any chance we can create a test case that merge can finish successfully?

File: node/lib/util/commit.js (hunk 2 above)
Comment (novalis) [reply node/lib/util/commit.js:565]: It passes the tests, and it didn't when I accidentally pushed mergeSubParent (without the .id()).  So. apparently, yes.

File: node/lib/util/commit.js (hunk 3 above)
Comment (novalis) [reply node/lib/util/commit.js:617]: I think git meta merge prevents you from starting a merge like that, but I can add an assert.

File: node/test/util/commit.js (hunk 4 above)
Comment (novalis) [reply node/test/util/commit.js:3419]: This is testing the case that you are mid-merge (that is, the merge has failed and you are now doing conflict resolution).

**Review by shijinglu — APPROVED** [review]

**Pull request merged**